Jellyfin.Server.Implementations/MediaSegments/MediaSegmentManager.cs

```diff
@@ -70,14 +70,29 @@ public async Task RunSegmentPluginProviders(BaseItem baseItem, LibraryOptions li
 
         using var db = await _dbProvider.CreateDbContextAsync(cancellationToken).ConfigureAwait(false);
 
-        if (!overwrite && (await db.MediaSegments.AnyAsync(e => e.ItemId.Equals(baseItem.Id), cancellationToken).ConfigureAwait(false)))
+        // delete all media segments from providers that have DeleteOnScan set.
         {
-            _logger.LogDebug("Skip {MediaPath} as it already contains media segments", baseItem.Path);
+            var deleteOnScanProviders = (await Task.WhenAll(providers
+                .Select(async e => (e, await e.DeleteOnScan))))
```

Comment:
this essentially amounts to an `async void` call which we should strongly avoid.
Use an foreach loop here instead